inc-complete-derive/src/lib.rs

```diff
@@ -0,0 +1,643 @@
+            )
+            .to_compile_error()
+            .into();
+        }
+    }
+
+    // Generate a call to impl_storage! macro
+    let expanded = quote! {
+        inc_complete::impl_storage!(#struct_name, #(#field_mappings),*);
+    };
+
+    TokenStream::from(expanded)
+}
+
+// =============================================================================
+// Input Derive Macro
+// =============================================================================
+
+/// Derive macro for Input computation
+///
+/// Usage:
+/// ```rust
+/// #[derive(Input)]
+/// #[inc_complete(id = 0, output = i32, storage = MyStorage)]
+/// struct MyInput;
+/// ```
+///
+/// This generates a call to `define_input!(0, MyInput -> i32, MyStorage)`
+/// which provides the actual implementation.
+///
+/// Required attributes:
+/// - `id`: The computation ID (integer)
+/// - `output`: The output type
+/// - `storage`: The storage type
+#[proc_macro_derive(Input, attributes(inc_complete))]
+pub fn derive_input(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+
+    let struct_name = &input.ident;
+
+    // Find the inc_complete attribute
+    let inc_complete_attr = input
+        .attrs
+        .iter()
+        .find(|attr| attr.path().is_ident("inc_complete"));
+
+    let attr = match inc_complete_attr {
+        Some(attr) => attr,
+        None => {
+            return syn::Error::new(
+                Span::call_site(),
+                "Input derive requires #[inc_complete(...)] attribute",
+            )
+            .to_compile_error()
+            .into();
+        }
+    };
+
+    // Parse the attribute arguments using the Input-specific parser
+    let mut parser = InputAttributeParser::new();
+    if let Err(err) = parser.parse_attribute_list(attr) {
+        return err.to_compile_error().into();
+    }
+
+    let (id, output_type, storage_type, assume_changed) = match parser.validate(attr.span()) {
+        Ok(values) => values,
+        Err(err) => return err.to_compile_error().into(),
+    };
+
+    // Generate the define_input! call
+    let expanded = if assume_changed {
+        quote! {
+            inc_complete::define_input!(#id, assume_changed #struct_name -> #output_type, #storage_type);
+        }
+    } else {
+        quote! {
+            inc_complete::define_input!(#id, #struct_name -> #output_type, #storage_type);
+        }
+    };
+
+    TokenStream::from(expanded)
+}
+
+// =============================================================================
+// Intermediate Macros
+// =============================================================================
+
+/// Procedural macro for defining intermediate computations directly on functions
+///
+/// Usage:
+/// ```rust
+/// // Assumes ComputedValue struct is already defined
+/// #[intermediate(id = 1)]
+/// fn compute_double(_computed: &ComputedValue, db: &DbHandle<MyStorage>) -> i32 {
```

Comment:
```suggestion
/// fn compute_double(_context: &ComputeDouble, db: &DbHandle<MyStorage>) -> i32 {
```
`_computed: &ComputedValue` makes it sounds like its the result of a computation instead of essentially the context struct providing any parameters to the function. Let's rename it to something like `_context: &ComputeDouble` or similar